src/kernels/kernelStartupCodeProvider.node.ts

```diff
@@ -2,85 +2,33 @@
-import { IWorkspaceService } from '../platform/common/application/types';
 import { CodeSnippets } from '../platform/common/constants';
 import { getFilePath } from '../platform/common/platform/fs-paths';
-import { IFileSystem } from '../platform/common/platform/types';
-import { IConfigurationService } from '../platform/common/types';
-import { calculateWorkingDirectory } from '../platform/common/utils.node';
 import { traceInfo } from '../platform/logging';
 import { isLocalHostConnection, isPythonKernelConnection } from './helpers';
-import { expandWorkingDir } from './jupyter/jupyterUtils';
+import { KernelWorkingFolder } from './kernelWorkingFolder.node';
 import { IKernel, isLocalConnection, IStartupCodeProvider, StartupCodePriority } from './types';
 
 @injectable()
 export class KernelStartupCodeProvider implements IStartupCodeProvider {
     public priority = StartupCodePriority.Base;
 
-    constructor(
-        @inject(IConfigurationService) private readonly configService: IConfigurationService,
-        @inject(IFileSystem) private readonly fs: IFileSystem,
-        @inject(IWorkspaceService) private readonly workspaceService: IWorkspaceService
-    ) {}
+    constructor(@inject(KernelWorkingFolder) private readonly kernelWorkingFolder: KernelWorkingFolder) {}
 
     async getCode(kernel: IKernel): Promise<string[]> {
-        // If this is a live kernel, we shouldn't be changing anything by running startup code.
+        // If this is a remote kernel, we shouldn't be changing the startup directory
         if (
-            !isPythonKernelConnection(kernel.kernelConnectionMetadata) &&
-            kernel.kernelConnectionMetadata.kind !== 'connectToLiveRemoteKernel'
+            !isLocalConnection(kernel.kernelConnectionMetadata) &&
+            !isLocalHostConnection(kernel.kernelConnectionMetadata) &&
+            !isPythonKernelConnection(kernel.kernelConnectionMetadata)
         ) {
             return [];
         }
-        if (
-            !(
-                isLocalConnection(kernel.kernelConnectionMetadata) &&
-                !this.configService.getSettings(undefined).forceIPyKernelDebugger
-            )
-        ) {
-            return [];
-        }
-
-        if (
-            isLocalConnection(kernel.kernelConnectionMetadata) ||
-            isLocalHostConnection(kernel.kernelConnectionMetadata)
-        ) {
-            let suggestedDir = await calculateWorkingDirectory(
-                this.configService,
-                this.workspaceService,
-                this.fs,
-                kernel.resourceUri
-            );
-            if (suggestedDir && (await this.fs.exists(suggestedDir))) {
-                traceInfo('UpdateWorkingDirectoryAndPath in Kernel');
-                // We should use the launch info directory. It trumps the possible dir
-                return this.getChangeDirectoryCode(kernel, suggestedDir);
-            } else if (kernel.resourceUri && (await this.fs.exists(kernel.resourceUri))) {
```

Comment:
1. We have logic here to calculate the working dir of a kernel (notice the if conditions and file path checks & joining of paths, etc, lots of stuff done here)